src/main.rs

```diff
@@ -6,11 +6,17 @@ mod features;
 fn main() -> Result<()> {
     let args = arg::parse_args();
 
-    println!("Cleaning directory: {:?}", args.dir);
-    println!("File types to clean: {:?}", args.types);
-    println!("Minimum file size: {} bytes", args.min_size);
-
-    features::cleaner::directory_cleaner(&args.dir, &args.types, args.min_size)?;
+    // precedence: file types, then file size. In case they both show up together in the command
+    // we'd do a refactor on here when we consider chaining
+    if args.types.len() > 0 {
+        println!("Cleaning directory: {:?}", args.dir);
+        println!("File types to clean: {:?}", args.types);
+        features::cleaner_file_type::directory_cleaner_based_on_file_type(&args.dir, &args.types)?;
+    } else if let Some(val) = args.min_size {
```

Comment:
I think we might be over abstracting these features
Why can't we use the existing logic of defaulting to 0 bytes if size isn't given ? 
This way we can simplify this code better and remove the dup codes here

Reply:
I don't think using zero as a default value is a great idea, an empty file is 0 bytes which could mean we have some file our code isn't considering just because they're empty.

I think it fits perfectly that we can make a field optional(using Option<T, E>) if they're not compulsory.